build.gradle.kts

```diff
@@ -117,46 +118,43 @@ tasks.javadoc {
 /* ******************** publishing ******************** */
 
 publishing {
-    publications.register<MavenPublication>("extensionSdk") {
-        from(components["java"])
+    publications {
+        create<MavenPublication>("extensionSdk") {
+            pom {
+                name.set(project.name)
+                description.set(project.description)
+                url.set(project.metadata.organization!!.url)
+                licenses {
+                    project.metadata.license
+                }
+                developers {
+                    project.metadata.developers
+                }
+                scm {
+                    connection.set("scm:git:git://github.com/hivemq/${project.metadata.github!!.repo}.git")
+                    developerConnection.set("scm:git:ssh://github.com/hivemq/${project.metadata.github!!.repo}.git")
+                    url.set("http://github.com/hivemq/${project.metadata.github!!.repo}/")
+                }
+            }
+        }
     }
 }
 
-bintray {
-    user = "${project.findProperty("bintrayUser") ?: System.getenv("BINTRAY_USER")}"
-    key = "${project.findProperty("bintrayApiKey") ?: System.getenv("BINTRAY_API_KEY")}"
-    dryRun = false
-    publish = false
-    override = false
-    pkg.apply {
-        userOrg = "hivemq"
-        repo = "HiveMQ"
-        name = "hivemq-extension-sdk"
-        desc = project.description
-        websiteUrl = metadata.url.get()
-        issueTrackerUrl = metadata.issueManagement!!.url.get()
-        vcsUrl = metadata.scm!!.url.get()
-        setLicenses(metadata.license!!.shortName.get())
-        setLabels("hivemq", "extension", "sdk", "mqtt", "mqtt5")
-        publicDownloadNumbers = false
-        version.apply {
-            released = Date().toString()
-            gpg.apply {
-                sign = true
-            }
-            mavenCentralSync.apply {
-                sync = false
-                user = "${project.findProperty("mavenCentralUser") ?: System.getenv("MAVEN_CENTRAL_USER")}"
-                password = "${project.findProperty("mavenCentralKey") ?: System.getenv("MAVEN_CENTRAL_KEY")}"
-                close = "0"
-            }
+nexusPublishing {
+    repositories {
+        sonatype {
+            username.set("${project.findProperty("sonatypeUser") ?: System.getenv("SONATYPE_USERNAME")}")
+            password.set("${project.findProperty("sonatypePassword") ?: System.getenv("SONATYPE_PASSWORD")}")
```

Comment:
should be unnecessary if you use the convention from the plugin
> you need to set the sonatypeUsername and sonatypePassword project properties, e.g. in ~/.gradle/gradle.properties or via the ORG_GRADLE_PROJECT_sonatypeUsername and ORG_GRADLE_PROJECT_sonatypePassword environment variables.

Reply:
Yes, but I want to use organisation wide Environment variables and want to keep it consistent over all projects.

Reply:
Of course you can name the env vars in GitHub/Jenkins whatever you want, just pass them to gradle in the right format

Reply:
Sticking to conventions is really important for build maintenance imho

Reply:
for example, you can assign different env names in the github workflow, or pass it to the gradle command like `ORG_GRADLE_PROJECT_sonatypeUsername=$SONATYPE_USERNAME ./gradlew ...` (I hope this is the correct syntax)

Reply:
Oh, I finally get your comment now! I thought the `ORG_GRADLE_PROJECT` is a specific prefix that has to change in every project. 

`ORG_GRADLE_PROJECT_something` makes gradle set the project property `something`. It has actually nothing to do with the plugin. 

🤦🏿